lib/Semantics/check-select-stmt.cpp

```diff
@@ -0,0 +1,398 @@
+//===-- lib/semantics/check-select-stmt.cc --------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "check-select-stmt.h"
+#include "flang/Parser/message.h"
+#include "flang/Semantics/tools.h"
+
+namespace Fortran::semantics {
+
+class SelectCaseHelper {
+
+public:
+  SelectCaseHelper(parser::Messages &messages) : messages_{messages} {}
+  void SetKind(int kind) { kind_ = kind; }
+  void SetTypeCategory(TypeCategory category) { category_ = category; }
```

Comment:
The coding standards call for this to be called `set_kind` and `set_category`. But hopefully you can eliminate these and pass the kind and category in to the constructor.

Reply:
I have modified to initialize these in constructor.